pymeasure/instruments/attocube/anc300.py

```diff
@@ -255,3 +247,65 @@ def stop_all(self):
             attribute = getattr(self, attr)
             if isinstance(attribute, Axis):
                 attribute.stop()
+
+    def __init__(self, host, axisnames, passwd, query_delay=0.05, **kwargs):
+        kwargs['query_delay'] = query_delay
+        kwargs['preprocess_reply'] = self.extract_value
+        address = f"TCPIP0::{host}::7230::SOCKET"
+        self.termination_str = "\r\n"
+
+        adapter = VISAAdapter(
+            address,
+            read_termination=self.termination_str,
+            write_termination=self.termination_str,
+            **kwargs
+        )
+
+        super().__init__(
+            adapter,
+            "attocube ANC300 Piezo Controller",
+            includeSCPI=False,
+            **kwargs
+        )
```

Comment:
Two points:
1. We require instruments to have a "name" parameter in the init signature.
2. You do not have to create the VISAAdapter manually. If you give a resource string to the super().__init__, a VISAAdapter will be created and kwargs are used for that creation.
3. We deprecated global usage of `preprocess_reply`, instead the corresponding code (here `_extract_reply`) can be put into the `read` method.

Here the implementation (except for moving it to the read method).
```suggestion
    def __init__(self, adapter, name="attocube ANC300 Piezo Controller", axisnames="", passwd="", query_delay=0.05, **kwargs):
        self.query_delay = query_delay
        self.termination_str = "\r\n"

        super().__init__(
            adapter,
            name,
            includeSCPI=False,
            read_termination=self.termination_str,
            write_termination=self.termination_str,
            **kwargs
        )
```

Reply:
Thanks, added.